sensors/ev3_uart_sensor_defs.c

```diff
@@ -86,25 +86,67 @@ const struct ev3_uart_sensor_info ev3_uart_sensor_defs[] = {
-				 * @description: Raw Reflected - sets LED color to red
-				 * @value0: ??? (0 to 1020???)
-				 * @value1: ??? (0 to 1020???)
+				 * .. [#lego-ev3-color-mode3-values] This mode
+				 *    provides direct access to the measurements
+				 *    done by the ADC inside the sensor. It
+				 *    works by continuously flashing the red LED
+				 *    on and off and measuring the current
+				 *    flowing through a bypass capacitor. It is
+				 *    not possible to measure true ambient light
+				 *    this way, but the LED off measurement
+				 *    works as a reference for the LED on
+				 *    measurement. Higher photodiode
+				 *    illumination will cause lower ADC
+				 *    measurements., so to obtain reflected
+				 *    light intensity similar to
+				 *    ``COL-REFLECT``, it is necessary to
+				 *    do ``reflect = value1 - value0``.
+				 *
+				 * @description: Raw Reflected Light - sets LED color to red
+				 * @value0: ADC reading for LED on (reflection) (0 to 1023)
+				 * @value0_footnote: [#lego-ev3-color-mode3-values]_
+				 * @value1: ADC reading for LED off (reference) (0 to 1023)
+				 * @value1_footnote: [#lego-ev3-color-mode3-values]_
 				 */
 				.name		= "REF-RAW",
 				.data_sets	= 2,
 				.data_type	= LEGO_SENSOR_DATA_S16,
+				.units		= "ADC",
 			},
 			[4] = {
 				/**
+				 * .. [#lego-ev3-color-mode4-value012] These
+				 *    values represent the difference between
+				 *    the raw sensor ADC measurements when the
+				 *    LEDs are off and when they are on. They
+				 *    work similar to the ``COL-REFLECT`` values
+				 *    but they are not precalibrated.
+				 *
+				 * .. [#lego-ev3-color-mode4-value3] This value
+				 *    is similar to the REF-RAW ambient
+				 *    ``value1``. It is the value measured by
+				 *    the ADC inside the sensor when all LEDs
+				 *    are off, so it works as a reference point
+				 *    for the first three values. However this
+				 *    time the ``value0``, ``value1`` and
+				 *    ``value2`` are sent relative to this value
+				 *    by the sensor automatically, so it is not
+				 *    necessary to use this value at all.
+				 *
 				 * @description: Raw Color Components - sets LED color to white (all LEDs rapidly cycling)
-				 * @value0: Red (0 to 1020???)
-				 * @value1: Green (0 to 1020???)
-				 * @value2: Blue (0 to 1020???)
-				 * @value3: Ambient? (0 to 1020???)
+				 * @value0: Reflected red light intensity (0 to 1023)
+				 * @value0_footnote: [#lego-ev3-color-mode4-value012]_
+				 * @value1: Reflected green light intensity (0 to 1023)
+				 * @value1_footnote: [#lego-ev3-color-mode4-value012]_
+				 * @value2: Reflected blue light intensity (0 to 1023)
+				 * @value2_footnote: [#lego-ev3-color-mode4-value012]_
+				 * @value3: ADC reading for LEDs off (reference) (0 to 1023)
+				 * @value3_footnote: [#lego-ev3-color-mode4-value3]_
 				 */
 				.name		= "RGB-RAW",
 				.data_sets	= 4,
 				.data_type	= LEGO_SENSOR_DATA_S16,
+				.units		= "ADC",
```

Comment:
Same here, this mode does not have units.

Reply:
Yup, I will remove it.